tensorflow_addons/optimizers/conditional_gradient.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Conditional Gradient method for TensorFlow."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+class ConditionalGradient(tf.keras.optimizers.Optimizer):
+    """Optimizer that implements the Conditional Gradient optimization.
+
+    This optimizer helps handle constraints well.
+
+    Currently only supports frobenius norm constraint.
+    See https://arxiv.org/pdf/1803.06453.pdf
+
+    ```
+    variable -= (1-learning_rate)
+        * (variable + l_ambda * gradient / frobenius_norm(gradient))
+    ```
+    """
+
+    def __init__(self,
+                 learning_rate,
+                 l_ambda,
```

Comment:
`l_ambda` doesn't follow name convention. I suggest to use `constraint` or `lambda_`, and explain it in the docstring. `constraint` looks better for me.

Reply:
I change the variable name into "lambda_". Thank you!